rust/lancedb/src/remote/table.rs

```diff
@@ -643,25 +643,85 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
     }
     async fn add_columns(
         &self,
-        _transforms: NewColumnTransform,
+        transforms: NewColumnTransform,
```

Comment:
Should we be using `lancedb::table::NewColumnTransform` instead of `lance::dataset::NewColumnTransform` (same question for some of the other types here)?